std/assembly/typedarray.ts

```diff
@@ -109,6 +114,10 @@ export class Uint8Array extends TypedArray<u8> {
   every(callbackfn: (value: u8, index: i32, self: Uint8Array) => bool): bool {
     return EVERY<Uint8Array, u8>(this, callbackfn);
   }
+
+  set<SourceT, SourceU>(source: SourceT, offset: i32 = 0): void {
```

Comment:
Also using `<SourceT, SourceU>` looks redundant. It should be `set<T>(source: T, offset: i32 = 0)`

Reply:
We cannot infer what the subtype is. It looks redundant, and is not fixable. Let me commit again and show you what I mean.

Reply:
If we want to use the `LOAD<T>` macro for arraybuffers, we must concede that we need a `SourceU` type.

There are many ways to do this in TypeScript, but AssemblyScript does not support this yet. Instead we need to declare the type up front. This is redundant, and feels wrong, but it's the only way to do this.

Reply:
I used a stub and instanceof. I don't know if this works well.